src/main/kotlin/no/nav/syfo/oppfolgingstilfelle/bit/database/OppfolginsgtilfelleBitQuery.kt

```diff
@@ -0,0 +1,87 @@
+package no.nav.syfo.oppfolgingstilfelle.bit.database
+
+import no.nav.syfo.application.database.DatabaseInterface
+import no.nav.syfo.application.database.toList
+import no.nav.syfo.domain.PersonIdentNumber
+import no.nav.syfo.oppfolgingstilfelle.bit.*
+import no.nav.syfo.oppfolgingstilfelle.bit.database.domain.POppfolgingstilfelleBit
+import no.nav.syfo.util.toOffsetDateTimeUTC
+import java.sql.*
+import java.util.*
+
+const val queryCreateOppfolgingstilfellebit =
+    """
+    INSERT INTO TILFELLE_BIT (
+        id,
+        uuid,
+        created_at,
+        inntruffet,
+        personident,
+        ressurs_id,
+        tags,
+        virksomhetsnummer,
+        fom,
+        tom
+        ) VALUES (DEFAULT, ?, ?, ?, ?, ?, ?, ?, ?, ?)
+        RETURNING id
+    """
+
+fun Connection.createOppfolgingstilfelleBit(
+    commit: Boolean,
+    oppfolgingstilfelleBit: OppfolgingstilfelleBit,
+) {
+    val oppfolgingstilfelleBitIdList = this.prepareStatement(queryCreateOppfolgingstilfellebit).use {
+        it.setString(1, oppfolgingstilfelleBit.uuid.toString())
+        it.setTimestamp(2, Timestamp.from(oppfolgingstilfelleBit.createdAt.toInstant()))
+        it.setTimestamp(3, Timestamp.from(oppfolgingstilfelleBit.inntruffet.toInstant()))
+        it.setString(4, oppfolgingstilfelleBit.personIdentNumber.value)
+        it.setString(5, oppfolgingstilfelleBit.ressursId)
+        it.setString(6, oppfolgingstilfelleBit.tagsToString())
+        it.setString(7, oppfolgingstilfelleBit.virksomhetsnummer)
+        it.setTimestamp(8, Timestamp.from(oppfolgingstilfelleBit.fom.toInstant()))
+        it.setTimestamp(9, Timestamp.from(oppfolgingstilfelleBit.tom.toInstant()))
+        it.executeQuery().toList { getInt("id") }
+    }
+
+    if (oppfolgingstilfelleBitIdList.size != 1) {
+        throw NoElementInsertedException("Creating TILFELLE_BIT failed, no rows affected.")
+    }
+
+    if (commit) {
+        this.commit()
+    }
+}
+
+const val queryGetOppfolgingstilfelleBitList =
+    """
+    SELECT *
+    FROM TILFELLE_BIT
+    WHERE personident = ?;
+    """
```

Comment:
Kanskje lurt med ORDER BY created_at DESC

(eller ASC avhengig av om lista skal ha de nyeste først eller sist)

Reply:
Created_at brukes ikke i koden for oppfolgingstilfelle nå. Kan legge til en sortering, f.eks på  fom og ASC, siden den er av datoene man ser på i koden.

Reply:
Kan legge til intruffet DESC inntil videre.